### PR TITLE
Add support for constraints in Ax generators using the Service API

### PR DESCRIPTION
**Pull request by AngelFP**
Adds the possibility of providing `parameter_constraints` and `outcome_constraints` in the generators that use the Ax Service API.

Example:
```python

var1 = VaryingParameter("x0", -50.0, 5.0)
var2 = VaryingParameter("x1", -5.0, 15.0)
obj = Objective("f", minimize=False)
p1 = Parameter("p1")

gen = AxSingleFidelityGenerator(
    varying_parameters=[var1, var2],
    objectives=[obj],
    analyzed_parameters=[p1],
    parameter_constraints=["x0 + x1 <= 10"],
    outcome_constraints=["p1 <= 30"]
)
```

### CODE REVIEW

File: tests/test_ax_generators.py [hunk 1]
```diff
@@ -74,9 +76,14 @@ def test_ax_single_fidelity():
     var1 = VaryingParameter("x0", -50.0, 5.0)
     var2 = VaryingParameter("x1", -5.0, 15.0)
     obj = Objective("f", minimize=False)
+    p1 = Parameter("p1")
 
     gen = AxSingleFidelityGenerator(
```
Comment (delaossa) [inline tests/test_ax_generators.py:81]: Constraint parameter capabilities were also added to `AxMultiFidelityGenerator`.
Don't you want to add a test for this too?

**Review by delaossa — COMMENTED** [review]
Thanks Angel. It looks good!

**Review by delaossa — APPROVED** [review]

File: tests/test_ax_generators.py (hunk 1 above)
Comment (AngelFP) [reply tests/test_ax_generators.py:83]: Thanks for the suggestion. Apparently the multi-fidelity generator does not accept `parameter_constraints` (see https://github.com/facebook/Ax/issues/1639), so I removed them. I updated the tests to include `outcome_constraints`.

Comment (delaossa) [reply tests/test_ax_generators.py:83]: Alright! Thanks!

**Pull request merged**